tests/test_modeling_common.py

```diff
@@ -642,13 +653,28 @@ def _create_and_check_torch_fx_tracing(self, config, inputs_dict, output_loss=Fa
                         sequence_length=sequence_length,
                         num_choices=num_choices,
                     )
-                    traced_output = traced_model(**prepared_inputs)
+                    traced_output = traced_model(**filtered_inputs)
 
             except RuntimeError:
                 self.fail("Couldn't trace module.")
 
             num_outputs = len(model_output)
+
+            def flatten_output(output):
+                flatten = []
+                for x in output:
+                    if isinstance(x, (tuple, list)):
+                        flatten += flatten_output(x)
+                    elif not isinstance(x, torch.Tensor):
+                        continue
+                    else:
+                        flatten.append(x)
+                return flatten
+
+            model_output = flatten_output(model_output)
+            traced_output = flatten_output(traced_output)
             outputs_are_close = all(torch.allclose(model_output[i], traced_output[i]) for i in range(num_outputs))
+
             self.assertTrue(outputs_are_close)
```

Comment:
1. I'm thinking that if this ever fails it'd be easier to debug if the loop were to be inverted:
```
for in in range(num_outputs):
    self.assert...
```
would it? I think trying to compare multiple tensors at once is much harder.

If you think it shouldn't really fail, then it is fine as it is.

2. but I'd just add a message to why this may fail, as otherwise it doesn't give context to the failure
```
self.assertTrue(outputs_are_close, "traced output doesn't match model_output")
```

I see that in many of our tests this context message is missing.